api/ceramic_cache/models.py

```diff
@@ -131,3 +134,110 @@ class Revocation(models.Model):
 
     def __str__(self):
         return f"Revocation #{self.pk}, proof_value={self.proof_value}"
+
+
+type BanType = Literal["account"] | Literal["hash"] | Literal["single_stamp"]
+
+
+class Ban(models.Model):
+    provider = models.CharField(default="", max_length=256, db_index=True, blank=True)
+    hash = models.CharField(default="", max_length=100, db_index=True, blank=True)
+    address = EthAddressField(default="", db_index=True, blank=True)
+    end_time = models.DateTimeField(null=True, blank=True)
+    reason = models.TextField(null=True, blank=True)
+    created_at = models.DateTimeField(auto_now_add=True)
+    last_run_revoke_matching = models.DateTimeField(
+        null=True, blank=True, help_text="Last time revoke_matching_credentials was run"
+    )
+
```

Comment:
I would include a `type` field, of type `BanType`.
That would be much clearer for people to understand, and query for particular ban types (I'm also thinking about analytics, not just eng)

Reply:
I'd prefer to restrict it to specific "types" at the UI level to keep the code simple. But I'm cool with whatever!